bigframes/display/anywidget.py

```diff
@@ -99,15 +97,17 @@ def __init__(self, dataframe: bigframes.dataframe.DataFrame):
         self._table_id = str(uuid.uuid4())
         self._all_data_loaded = False
         self._batch_iter: Optional[Iterator[pd.DataFrame]] = None
-        self._cached_batches: List[pd.DataFrame] = []
+        self._cached_batches: list[pd.DataFrame] = []
         self._last_sort_state: Optional[_SortState] = None
+        self._setting_html_lock = threading.RLock()
```

Comment:
Why use an RLock instead of a lock? Seems to me you only want one thread at a time running setting_html, not multiple.


```suggestion
        self._setting_html_lock = threading.Lock()
```

Reply:
RLock is needed because _set_table_html can be re-entrant when self.page is updated within the method, triggering the observer. I also added this information as comment